ci/test/00_setup_env_native_valgrind.sh

```diff
@@ -14,4 +14,6 @@ export NO_DEPENDS=1
 export TEST_RUNNER_EXTRA="--exclude feature_init,rpc_bind,feature_bind_extra"  # Excluded for now, see https://github.com/bitcoin/bitcoin/issues/17765#issuecomment-602068547
 export GOAL="install"
 # Temporarily pin dwarf 4, until using Valgrind 3.20 or later
-export BITCOIN_CONFIG="--enable-zmq --with-incompatible-bdb --with-gui=no CC=clang CXX=clang++ CFLAGS=-gdwarf-4 CXXFLAGS=-gdwarf-4"  # TODO enable GUI
+export BITCOIN_CONFIG="-DWITH_ZMQ=ON -DWARN_INCOMPATIBLE_BDB=OFF -DWITH_GUI=OFF \
+-DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ \
+-DCMAKE_C_FLAGS=-gdwarf-4 -DCMAKE_CXX_FLAGS=-gdwarf-4"  # TODO enable GUI
```

Comment:
Looking at this job, and native_tsan. Is there a reason CXXFLAGS are sometimes going into  the "env"`CXXFLAGS`, and sometimes into `DCMAKE_CXX_FLAGS` ?

Reply:
The former prepends generic non-configuration-specific flags being passed to the compiler. The latter overrides them.

Reply:
Right, but we do we need to preprend (what are actually cppflags) here. I don't see why they can't go in cxxflags?

Reply:
This diff mirrors the behavior from the master branch where the `CXXFLAGS` variable being set disables warning flags.

Reply:
Reworked. It is consistent with the `ci/test/00_setup_env_native_fuzz_with_valgrind.sh` now.